### DISCUSSION TITLE
yaml vs json for the kubernetes patch

### DISCUSSION THREAD

**Discussion by adrian-thurston** [flant/shell-operator#303] [Q&A] [answered]
Hi, this is a very nice project!

I'm curious if it's normal to send the commands to kubernetes in yaml or in json? Yaml is more natural for us because we do all our cluster configuration in yaml. However, there doesn't appear to be a jq equivalent installed in the shell-operator container. This would be handy for manipulating the output.

**Comment by diafour** [comment, marked as answer]
Hello! There is no "normal" way, just use what suits your needs. We use [mikefarah/yq](https://github.com/mikefarah/yq) utility to convert between JSON and YAML. You can install it in your Dockerfile (we do it in [deckhouse](https://github.com/deckhouse/deckhouse/blob/main/werf.yaml#L41)).

Also, I see the problem with an "object" field for "Create" patches. I think we should allow string with YAML definition in this field for JSON patches.

**Reply by diafour** [reply]
#308 will fix a problem with the "object" field.
